auto_update.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/blang/semver"
+
+	"github.com/inconshreveable/go-update"
+)
+
+// RunUpdater checks if an update is due, checks if current version is outdated and performs update if needed
+func RunUpdater(app *TGFApplication) bool {
+	var autoUpdateFile = "tgfautoupdate"
+	var dueForUpdate = lastRefresh(autoUpdateFile) > 2*time.Hour
+	if !dueForUpdate {
+		app.Debug("Update not due")
+		return false
+	}
+	touchImageRefresh(autoUpdateFile)
+
+	v, err := getLatestVersion()
+	if err != nil {
+		app.Debug("Unable to fetch latest version from S3", err)
+		return false
+	}
+
+	latestVersion, err := semver.Make(v)
+	if err != nil {
+		app.Debug("Semver error", err)
+		return false
+	}
+
+	currentVersion, err := semver.Make(version)
+	if err != nil {
+		app.Debug("Semver error", err)
+		return false
+	}
+
+	if !currentVersion.LT(latestVersion) {
+		app.Debug("Up to date", err)
+		return false
+	}
+
+	url := getPlatformZipURL(v)
+
+	if err := doUpdate(url); err != nil {
+		app.Debug("Failed update ! : %v", err)
+		return false
+	}
+
+	app.Debug("Updated")
```

Comment:
Should be visible to user

Reply:
@julienduchesne agree ?
https://github.com/coveooss/tgf/pull/79#discussion_r298232713
Maybe print a message, not a warning ?

Reply:
Yes my comment was only related to `Update not due` and `Tgf already up to date`. Sorry if that wasn't clear. Updating TGF is an extraordinary event and it should be displayed. You should also display the path of the executable as well as the old and new version if possible. Like this:
`Updated the executable at {path} from version {old} to {new}. The process has been restarted with the updated version.`

Reply:
I also think it should be a warning. A print doesn't catch the eye enough